README.md

```diff
@@ -47,3 +47,297 @@ Details for those wishing to develop the library can be found on the [wiki](http
 
 
 [![Build Status](https://travis-ci.org/prometheus/client_java.png?branch=master)](https://travis-ci.org/prometheus/client_java)
+
+## Instrumenting
+
+Four types of metric are offered: Counter, Gauge, Summary and Histogram.
+See the documentation on [metric types](http://prometheus.io/docs/concepts/metric_types/)
+and [instrumentation best practices](http://prometheus.io/docs/practices/instrumentation/#counter-vs.-gauge,-summary-vs.-histogram)
+on how to use them.
+
+### Counter
+
+Counters go up, and reset when the process restarts.
+
+
+```java
+import io.prometheus.client.Counter;
+class YourClass {
+  static final Counter requests = Counter.build()
+     .name("requests_total").help("Total requests.").register();
+  
+  void processRequest() {
+    requests.inc();
+    // Your code here.
+  }
+}
+```
+
+### Gauge
+
+Gauges can go up and down.
+
+```java
+class YourClass {
+  static final Gauge inprogressRequests = Gauge.build()
+     .name("inprogress_requests").help("Inprogress requests.").register();
+  
+  void processRequest() {
+    inprogressRequest.inc();
+    // Your code here.
+    inprogressRequest.dec();
+  }
+}
+```
+
+There are utilities for common use cases:
+
+```java
+gauge.setToCurrentTime(); //Set to current unixtime
+
+# Increment when entered, decrement when exited. (Not implemented yet)
+inprogressRequests.trackInProgress(() -> doWork());
+```
+
+A Gauge can also take its value from a callback:
+
+```java
+static final Gauge queueDepth = Gauge.build()
+     .name("widget_queue_depth").help("Widget queue depth.").register();
+
+void processRequest() {
+  queueDepth.setChild(new Child{
+      @Override
+      public double get() {
+          //executorService is a ThreadPoolExecutor
+          return executorService.getQueue().size();
+      }
+  })  
+}
+
+//or (Not implemented yet)
+queueDepth.setFunction(() -> executorService.getQueue().size());
+queueDepth.setFunction(() -> executorService.getQueue().size(), "label");
```

Comment:
I'd probably not mention setChild so prominently, as it may be used in appropriately. A brief mention that it exists for advanced use cases and a link to the javadoc should be enough.